Sources/HummingbirdCore/Request/ByteBufferStreamer.swift

```diff
@@ -184,12 +365,19 @@ public final class HBByteBufferStreamer: HBStreamerProtocol {
     /// - Returns: Returns an EventLoopFuture that will be fulfilled with array of ByteBuffers that has so far been fed to th request body
     ///     and whether we have consumed everything
     public func consume(on eventLoop: EventLoop) -> EventLoopFuture<HBStreamerOutput> {
-        if self.eventLoop.inEventLoop {
-            return self.consume().hop(to: eventLoop)
-        } else {
-            return self.eventLoop.flatSubmit {
-                self.consume()
-            }.hop(to: eventLoop)
+        self.state.runOnLoop { state, eventLoop in
+            state.consume(eventLoop: eventLoop)
+        }.hop(to: eventLoop)
```

Comment:
My only worry with these functions, old and new, is that the name shadowing of `eventLoop` is petty confusing to read.

Reply:
Those functions should be deprecated, all providing an event loop does if hop to that event loop at the end. I cold deprecate them in favour of the ones that don't provide an event loop